CryptoPrices/Sources/Data/Sources/Repositories/CoinRepository.swift

```diff
@@ -25,6 +26,14 @@ public class CoinRepository: CoinRepositoryProtocol {
         return try await coinAPI.trendingCoins(coinIDs: coinIDs).map { $0 as Coin }
     }
 
+    public func getChartPrices(coinID: String, filter: TimeFilter) async throws -> [ChartDataPoint] {
+        let chartData = try await coinAPI.getChartPrices(coinID: coinID, filter: filter)
+        let dataPoints = chartData.prices.map {
+            DataPoint(timestamp: $0.first ?? 0.0, price: $0.last ?? 0.0)
+        }
+        return dataPoints
+    }
+
```

Comment:
This is where we can transform from higher abstraction to lower one:
```swift
public func getChartPrices(coinID: String, filter: TimeFilter) async throws -> [ChartDataPoint] {
  return try await coinAPI.getChartPrices(coinID: coinID, numberOfDays: filter.daysCount).map { $0 as ChartDataPoint }
}

extension APIPrice: ChartDataPoint {}
```

It lies on the `numberOfDays: filter.daysCount` (`TimeFilter` to `Int`) 🎉

Reply:
As per the explanation from my [comment](https://github.com/nimblehq/nimble-crypto-ios/pull/75#discussion_r1058163342), I don't see any major concern to stop using enum here. Furthermore, looking at your suggestion, I think we should try to avoid complex operations as per our [convention](https://nimblehq.co/compass/development/ios/build-time-optimization/) here, and thus, we should create a `let` constant per each operation instead. 🙏